### PR TITLE
try micromamba

### PR DESCRIPTION
**Pull request by ocefpaf**
Debugging #1186

### CODE REVIEW

**Comment by ocefpaf** [comment]
@hmaarrfk I believe the changes in https://github.com/conda-forge/zlib-feedstock/pull/62 broke the Windows CIs here. Using the previous version or build number before that PR works. As the author of that PR do you mind taking a look here and help us figure this out?

@jswhit this PR adds a workaround for the zlib issue but we should unpin it when the problem is fixed upstream. I'll leave to you if we should merge this or not. There are other changes that are worth keeping regardless of decision on the zlib pin.

**Comment by ocefpaf** [comment]
I believe the failure there is a network issue and should pass with restart.

**Comment by hmaarrfk** [comment]
https://github.com/conda-forge/libnetcdf-feedstock/issues/146

**Comment by hmaarrfk** [comment]
is the issue that libnetcdf fails, or that this build here fails?

**Comment by ocefpaf** [comment]
> is the issue that libnetcdf fails, or that this build here fails?

The build here fails. The debug info is in https://github.com/Unidata/netcdf4-python/issues/1186#issue-1357408267

**Comment by jswhit** [comment]
thanks for fixing this so quickly @ocefpaf !

**Comment by hmaarrfk** [comment]
do you have a build that just add `zlib` instead of pinning it?

File: .github/workflows/miniconda.yml [hunk 1]
```diff
@@ -18,29 +18,35 @@ jobs:
...
-        activate-conda: false
-        conda-channels: conda-forge
+        environment-file: false
 
     - name: Python ${{ matrix.python-version }}
       shell: bash -l {0}
       run: |
-        conda create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest hdf5 libnetcdf cftime --strict-channel-priority
-        source activate TEST
+        micromamba create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest hdf5 libnetcdf cftime "zlib=1.2.11" --channel conda-forge
```
Comment (hmaarrfk) [inline .github/workflows/miniconda.yml:34]:
```suggestion
        micromamba create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest hdf5 libnetcdf cftime zlib --channel conda-forge
```
does this work?

File: .github/workflows/miniconda.yml [hunk 2]
```diff
@@ -53,26 +59,30 @@ jobs:
...
-        activate-conda: false
-        conda-channels: conda-forge
+        environment-file: false
 
     - name: Python ${{ matrix.python-version }}
       shell: bash -l {0}
       run: |
-        conda create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest mpi4py hdf5=*=mpi* libnetcdf=*=mpi* cftime --strict-channel-priority
-        source activate TEST
+        micromamba create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest mpi4py hdf5=*=mpi* libnetcdf=*=mpi* cftime --channel conda-forge
```
Comment (hmaarrfk) [inline .github/workflows/miniconda.yml:70]:
```suggestion
        micromamba create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest mpi4py hdf5=*=mpi* libnetcdf=*=mpi* cftime zlib --channel conda-forge
```

File: .github/workflows/miniconda.yml (hunk 1 above)
Comment (hmaarrfk) [reply .github/workflows/miniconda.yml:34]: Deeper explanation; https://github.com/Unidata/netcdf4-python/issues/1186#issuecomment-1233339986

**Comment by hmaarrfk** [comment]
feel free to squash.

**Comment by ocefpaf** [comment]
> feel free to squash.

I'll leave that to @jswhit, not sure what he prefers. Thanks @hmaarrfk for the quick debug!

**Comment by jswhit** [comment]
If the fix upstream will happen soon, I'd prefer to wait for that.  I'll leave this open in the meantime.  Which changes in this PR do we need even after the upstream fix?

**Comment by ocefpaf** [comment]
> If the fix upstream will happen soon, I'd prefer to wait for that. I'll leave this open in the meantime. Which changes in this PR do we need even after the upstream fix?

Sorry. I should've said before. @hmaarrfk figured out the real issue and it is that we were depending on an indirect pull of `libzlib`. B/c netcdf4-python depends on zlib directly we have to make it an explicit dependency, no need for workarounds with previous versions.

This should be good to go as-is.

**Pull request merged**